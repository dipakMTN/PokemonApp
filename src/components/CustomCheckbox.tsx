import React from 'react';
import { TouchableOpacity, View, Text, StyleSheet } from 'react-native';

interface CustomCheckboxProps {
  isChecked: boolean;
  onPress: () => void;
}

const CustomCheckbox: React.FC<CustomCheckboxProps> = ({ isChecked, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <View style={[styles.checkbox, isChecked && styles.checked]}>
        {isChecked && <Text style={styles.checkmark}>âœ“</Text>}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  checkbox: {
    width: 24,
    height: 24,
    borderWidth: 1,
    borderColor: '#007AFF',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 8,
  },
  checked: {
    backgroundColor: '#007AFF',
  },
  checkmark: {
    color: '#fff',
    fontSize: 16,
  },
});

export default CustomCheckbox;
