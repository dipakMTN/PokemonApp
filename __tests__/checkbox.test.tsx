import React from 'react';
import renderer from 'react-test-renderer';
import CustomCheckbox from '../src/components/checkbox/checkbox';

// Mocking react-native-responsive-fontsize
jest.mock('react-native-responsive-fontsize', () => ({
  RFPercentage: jest.fn((val) => val),
  RFValue: jest.fn((val) => val),
}));

describe('CustomCheckbox', () => {
  test('renders unchecked checkbox', () => {
    const tree = renderer
      .create(<CustomCheckbox isChecked={false} onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders checked checkbox', () => {
    const tree = renderer
      .create(<CustomCheckbox isChecked={true} onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calls onPress callback when pressed', () => {

    // TouchableOpacity component could not be found in the rendered component tree
    // So, need to direct take reference of component by rendering it

    const onPressMock = jest.fn();
    const component = renderer.create(
      <CustomCheckbox isChecked={false} onPress={onPressMock} />,
    );
    const instance = component.root;
    instance.props.onPress(); // Call the onPress handler directly
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
});
