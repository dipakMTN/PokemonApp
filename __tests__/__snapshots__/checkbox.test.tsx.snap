// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckbox renders checked checkbox 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 24,
        },
        {
          "backgroundColor": "#007AFF",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      âœ“
    </Text>
  </View>
</View>
`;

exports[`CustomCheckbox renders unchecked checkbox 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 24,
        },
        false,
      ]
    }
  />
</View>
`;
